### PR TITLE
Add e2e pipeline and fixes

### PR DESCRIPTION
**Pull request by reakaleek**
Add e2e test.

A dynamic pipeline that uses the plugin with the current commit.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+
 steps:
-  - label: "E2E Test"
-    command: echo "TODO"
+  - label: ":buildkite:	Upload E2E pipeline"
+    if: build.branch == 'main' || build.source == "ui"
+    env:
+      OBLT_CLI_VERSION: 7.3.0
+    command: |
+      buildkite-agent pipeline upload <<-YAML
```
Comment (v1v) [inline .buildkite/pipeline.yml:10]: What's the reason for using the dynamic approach?

Comment (v1v) [reply .buildkite/pipeline.yml:10]: aha... I said nothing

Comment (reakaleek) [reply .buildkite/pipeline.yml:10]: I didn't know how to provide the current commit as plugin version in `- elastic/oblt-cli#<version>`.

Or am I overthinking this?

Comment (v1v) [reply .buildkite/pipeline.yml:10]: wait... cannot you use it without using the dynamic approach?

Comment (reakaleek) [reply .buildkite/pipeline.yml]: Yup. I was overthinking it.

It works.. https://github.com/elastic/oblt-cli-buildkite-plugin/pull/2/commits/8bc9ed553240cee70a7506746c0724bfea1814c7

Comment (reakaleek) [reply .buildkite/pipeline.yml]: Thank you for the hint!

**Review by v1v — APPROVED** [review]

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+      GITHUB_TOKEN: ${VAULT_GITHUB_TOKEN}
+
+  - wait: ~
+
+  - label: "E2E - Get cluster name"
+    command: buildkite-agent meta-data get cluster-name
+
+notify:
+  - slack: "#observablt-bots"
+    if: build.branch == "main" && build.state == "failed" && build.source != "trigger_job"
```
Comment (reakaleek) [inline .buildkite/pipeline.yml:19]:
```suggestion
    if: build.branch == "main" && build.state == "failed"
```

File: catalog-info.yaml [hunk 3]
```diff
@@ -24,6 +24,8 @@ spec:
     spec:
       repository: elastic/oblt-cli-buildkite-plugin
       pipeline_file: ".buildkite/pipeline.yml"
+      provider_settings:
+        build_pull_requests: false
```
Comment (reakaleek) [inline catalog-info.yaml:28]:
```suggestion
```

**Pull request merged**